### ISSUE TITLE
loss functions with data-independent terms

### ISSUE THREAD

**Issue by dionman** [SsnL/PyTorch-Reparam-Module#5]
Can this module accommodate loss functions (having components) that involve the neural network parameters, but are not functions of input data, e.g. a regularizer or a prior on the model parameters?

**Comment by ssnl** [comment]
Yes I believe you can do that.

**Comment by dionman** [comment]
Using your example, let's say we also have a L2 regularizer on the network weights, i.e.
```loss = F.cross_entropy(reparam_net(train_data, flat_param=theta), train_label) + sum(sum(net.parameters()))```. What's the recommended way to write down the cose?
